reflection/serverreflection_test.go

```diff
@@ -570,3 +555,43 @@ func testListServices(t *testing.T, stream rpb.ServerReflection_ServerReflection
 		t.Errorf("ListServices = %v, want type <ServerReflectionResponse_ListServicesResponse>", r.MessageResponse)
 	}
 }
+
+func registerDynamicProto(srv *grpc.Server, fd *dpb.FileDescriptorProto) {
+	df, err := protodesc.NewFile(fd, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	err = protoregistry.GlobalFiles.RegisterFile(df)
```

Comment:
This and `RegisterMessage` cause panic because they might register duplicate entries to the global registry. You might want to move those `protoregistry.GlobalXXX.RegisterXXX` to `init()`.

You can test this by `go test . -v -cpu 1,4